### PR TITLE
[ADD] sale_backorder

### PR DESCRIPTION
**Pull request by max3903**
#68

Previously known as osi_so_backorder_report

### CODE REVIEW

**Comment by max3903** [comment]
Ping @SerenaCyr

**Review by osimallen — APPROVED** [review]
Looks good

**Review by smangukiya — APPROVED** [review]

**Comment by max3903** [comment]
@murtuzasaleh Can you fix the Runbot warning?

WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (product_type, prod_type) of sale.order.line() have the same label: Product Type.

File: sale_backorder/models/sale.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+                          compute='_compute_bo_qty', store=True)
+    uigd_value = fields.Monetary(string="UIGD Value",
+                                 compute='_compute_uigd_value', store=True)
+    bo_value = fields.Monetary(string="Backorder Value",
+                               compute='_compute_bo_value', store=True)
+    prod_type = fields.Selection(string="Product Type",
+                                 selection=PRODUCT_TYPE_SELECTION,
+                                 related="product_id.type")
+    product_type = fields.Selection(string='Product Type',
+                                    related='product_id.product_tmpl_id.type')
```
Comment (max3903) [inline sale_backorder/models/sale.py:87]: @smangukiya Why do we need 2 fields here? The template and the product always have the same type.

**Comment by murtuzasaleh** [comment]
> @murtuzasaleh Can you fix the Runbot warning?
> 
> WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (product_type, prod_type) of sale.order.line() have the same label: Product Type.

Done, Improved code, report and fixed `So Backorder Report` error

so

**Comment by murtuzasaleh** [comment]
@max3903 Can you please review this PR.

**Pull request merged**